tools/driver/Driver.cpp

```diff
@@ -612,47 +612,51 @@ int Driver::MainLoop() {
   // are processed.
   WriteCommandsForSourcing(eCommandPlacementBeforeFile, commands_stream);
 
-  const size_t num_args = m_option_data.m_args.size();
-  if (num_args > 0) {
-    char arch_name[64];
-    if (lldb::SBDebugger::GetDefaultArchitecture(arch_name, sizeof(arch_name)))
-      commands_stream.Printf("target create --arch=%s %s", arch_name,
-                             EscapeString(m_option_data.m_args[0]).c_str());
-    else
-      commands_stream.Printf("target create %s",
-                             EscapeString(m_option_data.m_args[0]).c_str());
-
-    if (!m_option_data.m_core_file.empty()) {
-      commands_stream.Printf(" --core %s",
-                             EscapeString(m_option_data.m_core_file).c_str());
-    }
-    commands_stream.Printf("\n");
+  // If we're not in --repl mode, add the commands to process the file
+  // arguments, and the commands specified to run afterwards.
+  if (!m_option_data.m_repl) {
```

Comment:
Definitely want to make sure an LLDB person signs off on this.

Reply:
@jimingham I wanted to get your opinon on this, too. Currently`lldb --repl` always ignores `-O` options, but for testing they would be really useful.

Nathan made sure that the Swift driver always passes `-x` to `lldb --repl` to preserve the old behavior for users of `swift` but if you call `lldb --repl` you can now also specify `-O` options.

Reply:
I think there was some confusion here, sorry. I *was* going to do that, but then I realized the existing behavior with `--repl` (and from swift) was actually already processing the .lldbinit from the home directory (it does so a little earlier in this function, before any of the command_stream/repl code changed here) and that behavior is actually useful as it gives us a way to set the new `swift-module-loading-mode` setting when running the repl via swift. I think this PR will change the behavior from swift, but only for users who have lldbinit files in their project directories and whose home directory lldbinit file sets the `target.load-cwd-lldbinit` setting to true (the default is to ignore them warn that they exist).